### PR TITLE
RFC: Add Code Analyzers

### PR DESCRIPTION
**Pull request by RLittlesII**
Associated to #18

### CODE REVIEW

File: reactiveui/0002-code-analyzers.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+2. We use four spaces of indentation (no tabs).
+3. We use `_camelCase` for internal and private fields and use `readonly` where possible. Prefix internal and private instance fields with `_`, static fields with `s_` and thread static fields with `t_`. When used on static fields, `readonly` should come after `static` (e.g. `static readonly` not `readonly static`).  Public fields should be used sparingly and should use PascalCasing with no prefix when used.
+4. We avoid `this.` unless absolutely necessary.
+5. We always specify the visibility, even if it's the default (e.g.
+   `private string _foo` not `string _foo`). Visibility should be the first modifier (e.g.
+   `public abstract` not `abstract public`).
+6. Namespace imports should be specified at the top of the file, *outside* of
```
Comment (glennawatson) [inline reactiveui/0002-code-analyzers.md]: I guess we do System ones first (alphabetically), then other using's again alphabetically but separate to the Sytem's orderings.

File: reactiveui/0002-code-analyzers.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+        public int CountC# Coding Style
+===============
+
+For non code files (xml, etc), our current best guidance is consistency. When editing files, keep new code and changes consistent with the style in the files. For new files, it should conform to the style for that component. If there is a completely new component, anything that is reasonably broadly accepted is fine.
+
+The general rule we follow is "use Visual Studio defaults".
+
+1. We use [Allman style](http://en.wikipedia.org/wiki/Indent_style#Allman_style) braces, where each brace begins on a new line. A single line statement block can go without braces but the block must be properly indented on its own line and must not be nested in other statement blocks that use braces (See issue [381](https://github.com/dotnet/corefx/issues/381) for examples). One exception is that a `using` statement is permitted to be nested within another `using` statement by starting on the following line at the same indentation level, even if the nested `using` contains a controlled block.
+2. We use four spaces of indentation (no tabs).
```
Comment (glennawatson) [inline reactiveui/0002-code-analyzers.md]: Is this a repeat of what's up above?

**Review by glennawatson — CHANGES_REQUESTED** [review]

**Comment by glennawatson** [comment]
When you're done with getting this in, we should probably get this out to the website as well.

File: reactiveui/0002-code-analyzers.md (hunk 1 above)
Comment (RLittlesII) [reply reactiveui/0002-code-analyzers.md]: Agreed.  Updated verbiage to denote.

File: reactiveui/0002-code-analyzers.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
+C# Coding Style
+===============
+
+The general rule we follow is "use Visual Studio defaults".
+
+1. We use [Allman style](http://en.wikipedia.org/wiki/Indent_style#Allman_style) braces, where each brace begins on a new line. A single line statement block can go without braces but the block must be properly indented on its own line and must not be nested in other statement blocks that use braces (See RFC [18](https://github.com/reactiveui/rfcs/issues/18) for examples). One exception is that a `using` statement is permitted to be nested within another `using` statement by starting on the following line at the same indentation level, even if the nested `using` contains a controlled block.
+2. We use four spaces of indentation (no tabs).
+3. We use `_camelCase` for internal and private fields and use `readonly` where possible. Prefix internal and private instance fields with `_`, static fields with `s_` and thread static fields with `t_`. When used on static fields, `readonly` should come after `static` (e.g. `static readonly` not `readonly static`).  Public fields should be used sparingly and should use PascalCasing with no prefix when used.
```
Comment (glennawatson) [inline reactiveui/0002-code-analyzers.md]: Don't think we do any of the s_ and t_ stuff in our code base.

**Review by glennawatson — CHANGES_REQUESTED** [review]

**Review by glennawatson — APPROVED** [review]

**Pull request merged**